### PR TITLE
Dispalying the 4th derivative correctly

### LINKED ISSUE #10173: mechanics_printing module does not display 4th derivatives correctly
When I run the following code snippet in an IPython notebook, the 1st, 2nd, and 3rd derivatives display correctly, but the 4th derivative does not. The 4th derivative simply displays as p, rather than as \ddddot{p}.

```
%pylab inline

import sympy.physics.mechanics.functions

sympy.physics.mechanics.functions.mechanics_printing(use_latex="mathjax", latex_mode="equation")

t = sympy.Symbol("t")
p = sympy.Symbol("p")(t)

p, p.diff(t), p.diff(t).diff(t), p.diff(t).diff(t).diff(t), p.diff(t).diff(t).diff(t).diff(t)
```

At first glance, this behavior seems to be caused by the `_print_Derivative` function in sympy/sympy/physics/vector/printing.py not handling 4th derivatives. I'll try hacking this function locally to support 4th derivatives and report back if this works around the issue.

Ideally, mechanics_printing should gracefully fall back to Leibniz notation in the case of high-order derivatives, rather than printing them incorrectly.

### PR DESCRIPTION
**Pull request by kshitij10496**
fixes #10173 

Ping @moorepants

### CODE REVIEW

**Comment by moorepants** [comment]
This looks fine but you should add a unit test for the new functionality.

**Comment by kshitij10496** [comment]
@moorepants Should I add the test [here](https://github.com/sympy/sympy/blob/master/sympy/physics/vector/tests/test_printing.py#L120) ?

**Comment by moorepants** [comment]
Yes.

**Comment by kshitij10496** [comment]
Cool.
Onto it

**Comment by kshitij10496** [comment]
@moorepants This doesn't fallback as expected.
I am getting an error:
`AttributeError: 'super' object has no attribute '_print_derivative'`

How should I go about fixing this ?

**Comment by moorepants** [comment]
I don't know you'll have to post the code you typed and the full traceback. You also need to add a unit test for this.

**Comment by kshitij10496** [comment]
Here is the completely traceback from an IPython Notebook:
```
In [ ]: p.diff().diff().diff().diff()
Out[ ]: p⃜ 
In [ ]: p.diff().diff().diff().diff().diff()
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
/home/kshitij10496/anaconda2/lib/python2.7/site-packages/IPython/core/formatters.pyc in __call__(self, obj)
    695                 type_pprinters=self.type_printers,
    696                 deferred_pprinters=self.deferred_printers)
--> 697             printer.pretty(obj)
    698             printer.flush()
    699             return stream.getvalue()

/home/kshitij10496/anaconda2/lib/python2.7/site-packages/IPython/lib/pretty.pyc in pretty(self, obj)
    366                 if cls in self.type_pprinters:
    367                     # printer registered in self.type_pprinters
--> 368                     return self.type_pprinters[cls](obj, self, cycle)
    369                 else:
    370                     # deferred printer

/home/kshitij10496/Repo/sympy/sympy/interactive/printing.pyc in _print_plain(arg, p, cycle)
     66         """caller for pretty, for use in IPython 0.11"""
     67         if _can_print_latex(arg):
---> 68             p.text(stringify_func(arg))
     69         else:
     70             p.text(IPython.lib.pretty.pretty(arg))

/home/kshitij10496/Repo/sympy/sympy/physics/vector/printing.py in vpprint(expr, **settings)
    334 
    335     try:
--> 336         return pp.doprint(expr)
    337     finally:
    338         pretty_use_unicode(uflag)

/home/kshitij10496/Repo/sympy/sympy/printing/pretty/pretty.py in doprint(self, expr)
     56 
     57     def doprint(self, expr):
---> 58         return self._print(expr).render(**self._settings)
     59 
     60     # empty op so _print(stringPict) returns the same

/home/kshitij10496/Repo/sympy/sympy/printing/printer.pyc in _print(self, expr, *args, **kwargs)
    255                 printmethod = '_print_' + cls.__name__
    256                 if hasattr(self, printmethod):
--> 257                     return getattr(self, printmethod)(expr, *args, **kwargs)
    258 
    259             # Unknown object, fall back to the emptyPrinter.

/home/kshitij10496/Repo/sympy/sympy/physics/vector/printing.py in _print_Derivative(self, deriv)
    201         uni_split = [uni[:lu], uni[lu:]]
    202 
--> 203         d['picture'] = [pic_split[0] + dots[dot_i] + pic_split[1]]
    204         d['unicode'] =  uni_split[0] + dots[dot_i] + uni_split[1]
    205 

KeyError: 5

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/home/kshitij10496/anaconda2/lib/python2.7/site-packages/IPython/core/formatters.pyc in __call__(self, obj)
    335                 pass
    336             else:
--> 337                 return printer(obj)
    338             # Finally look for special method names
    339             method = _safe_get_formatter_method(obj, self.print_method)

/home/kshitij10496/Repo/sympy/sympy/interactive/printing.pyc in _print_latex_text(o)
    149         """
    150         if _can_print_latex(o):
--> 151             s = latex(o, mode='plain')
    152             s = s.replace(r'\dag', r'\dagger')
    153             s = s.strip('$')

/home/kshitij10496/Repo/sympy/sympy/physics/vector/printing.py in vlatex(expr, **settings)
    376     latex_printer = VectorLatexPrinter(settings)
    377 
--> 378     return latex_printer.doprint(expr)
    379 
    380 

/home/kshitij10496/Repo/sympy/sympy/printing/latex.py in doprint(self, expr)
    152 
    153     def doprint(self, expr):
--> 154         tex = Printer.doprint(self, expr)
    155 
    156         if self._settings['mode'] == 'plain':

/home/kshitij10496/Repo/sympy/sympy/printing/printer.pyc in doprint(self, expr)
    231     def doprint(self, expr):
    232         """Returns printer's representation for expr (as a string)"""
--> 233         return self._str(self._print(expr))
    234 
    235     def _print(self, expr, *args, **kwargs):

/home/kshitij10496/Repo/sympy/sympy/printing/printer.pyc in _print(self, expr, *args, **kwargs)
    255                 printmethod = '_print_' + cls.__name__
    256                 if hasattr(self, printmethod):
--> 257                     return getattr(self, printmethod)(expr, *args, **kwargs)
    258 
    259             # Unknown object, fall back to the emptyPrinter.

/home/kshitij10496/Repo/sympy/sympy/physics/vector/printing.py in _print_Derivative(self, der_expr)
    152             base = r"\ddddot{%s}" % base
    153         else:
--> 154             return super(VectorLatexPrinter, self)._print_derivative(der_expr)
    155         if len(base_split) is not 1:
    156             base += '_' + base_split[1]

AttributeError: 'super' object has no attribute '_print_derivative'

```

**Comment by moorepants** [comment]
Does the super class have the needed method?

**Comment by kshitij10496** [comment]
No. It doesn't.

File: sympy/physics/vector/printing.py [hunk 1]
```diff
@@ -148,6 +148,10 @@ def _print_Derivative(self, der_expr):
             base = r"\ddot{%s}" % base
         elif dots == 3:
             base = r"\dddot{%s}" % base
+        elif dots == 4:
+            base = r"\ddddot{%s}" % base
+        else:
+            return super(VectorLatexPrinter, self)._print_derivative(der_expr)
```
Comment (moorepants) [inline sympy/physics/vector/printing.py]: Is it just supposed to be `_print_Derivative`?

**Comment by moorepants** [comment]
You'll need to add the correct methods to the super class or something here.

File: sympy/physics/vector/printing.py (hunk 1 above)
Comment (kshitij10496) [inline sympy/physics/vector/printing.py]: @moorepants Yeah. I am sorry.
This was a typo perhaps.

**Comment by kshitij10496** [comment]
```
In []: %pylab inline
​
import sympy.physics.mechanics.functions
​
sympy.physics.mechanics.functions.mechanics_printing(use_latex="mathjax", latex_mode="equation")
​
t = sympy.Symbol("t")
p = sympy.Symbol("p")(t)
​
p, p.diff(t), p.diff(t).diff(t), p.diff(t).diff(t).diff(t), p.diff(t).diff(t).diff(t).diff(t), p.diff(t).diff(t).diff(t).diff(t).diff(t)
​
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
/home/kshitij10496/anaconda2/lib/python2.7/site-packages/IPython/core/formatters.pyc in __call__(self, obj)
    695                 type_pprinters=self.type_printers,
    696                 deferred_pprinters=self.deferred_printers)
--> 697             printer.pretty(obj)
    698             printer.flush()
    699             return stream.getvalue()

/home/kshitij10496/anaconda2/lib/python2.7/site-packages/IPython/lib/pretty.pyc in pretty(self, obj)
    366                 if cls in self.type_pprinters:
    367                     # printer registered in self.type_pprinters
--> 368                     return self.type_pprinters[cls](obj, self, cycle)
    369                 else:
    370                     # deferred printer

/home/kshitij10496/Repo/sympy/sympy/interactive/printing.pyc in _print_plain(arg, p, cycle)
     66         """caller for pretty, for use in IPython 0.11"""
     67         if _can_print_latex(arg):
---> 68             p.text(stringify_func(arg))
     69         else:
     70             p.text(IPython.lib.pretty.pretty(arg))

/home/kshitij10496/Repo/sympy/sympy/physics/vector/printing.py in vpprint(expr, **settings)
    334 
    335     try:
--> 336         return pp.doprint(expr)
    337     finally:
    338         pretty_use_unicode(uflag)

/home/kshitij10496/Repo/sympy/sympy/printing/pretty/pretty.py in doprint(self, expr)
     56 
     57     def doprint(self, expr):
---> 58         return self._print(expr).render(**self._settings)
     59 
     60     # empty op so _print(stringPict) returns the same

/home/kshitij10496/Repo/sympy/sympy/printing/printer.pyc in _print(self, expr, *args, **kwargs)
    255                 printmethod = '_print_' + cls.__name__
    256                 if hasattr(self, printmethod):
--> 257                     return getattr(self, printmethod)(expr, *args, **kwargs)
    258 
    259             # Unknown object, fall back to the emptyPrinter.

/home/kshitij10496/Repo/sympy/sympy/printing/pretty/pretty.py in _print_tuple(self, t)
   1648             return prettyForm(*ptuple.parens('(', ')', ifascii_nougly=True))
   1649         else:
-> 1650             return self._print_seq(t, '(', ')')
   1651 
   1652     def _print_Tuple(self, expr):

/home/kshitij10496/Repo/sympy/sympy/printing/pretty/pretty.py in _print_seq(self, seq, left, right, delimiter, parenthesize)
   1608 
   1609         for item in seq:
-> 1610             pform = self._print(item)
   1611 
   1612             if parenthesize(item):

/home/kshitij10496/Repo/sympy/sympy/printing/printer.pyc in _print(self, expr, *args, **kwargs)
    255                 printmethod = '_print_' + cls.__name__
    256                 if hasattr(self, printmethod):
--> 257                     return getattr(self, printmethod)(expr, *args, **kwargs)
    258 
    259             # Unknown object, fall back to the emptyPrinter.

/home/kshitij10496/Repo/sympy/sympy/physics/vector/printing.py in _print_Derivative(self, deriv)
    201         uni_split = [uni[:lu], uni[lu:]]
    202 
--> 203         d['picture'] = [pic_split[0] + dots[dot_i] + pic_split[1]]
    204         d['unicode'] =  uni_split[0] + dots[dot_i] + uni_split[1]
    205 

KeyError: 5

Out[5]:
(p,p˙,p¨,p⃛ ,p⃜ ,d5dt5p)
```

**Comment by kshitij10496** [comment]
@moorepants 
I am getting a `KeyError` here.
Can you validate this and kindly suggest some fix for this ?

**Comment by moorepants** [comment]
Does unicode support five dots?

**Comment by kshitij10496** [comment]
I don't think so.

I am getting the same `KeyError` with an identical traceback as in the above case.

**Comment by moorepants** [comment]
So, if you call the unicode pretty printing with five dots it should also fall back to Leibniz notation.

**Comment by kshitij10496** [comment]
Ideally, it should.

But apparently, its not falling back to Leibniz notation

**Comment by moorepants** [comment]
Then you should either fix it here or raise a new issue to fix it in the future. It should be simple to fix, just like you've done with the LaTeX printer.

**Comment by kshitij10496** [comment]
I will try to fix it.
Thanks

**Comment by moorepants** [comment]
This still needs a unit test.

**Comment by oscarbenjamin** [comment]
Closing as superseded by #15849

**Pull request closed without merging**